src/reader/ns_reader.rs

```diff
@@ -189,6 +255,121 @@ impl<'i> NsReader<&'i [u8]> {
+    ///                 b"tag2" => assert_eq!(ns, Bound(Namespace(b"www.yyyy"))),
+    ///                 _ => unreachable!(),
+    ///             }
+    ///         }
+    ///         Event::Text(e) => {
+    ///             txt.push(e.decode_and_unescape(&reader).unwrap().into_owned())
+    ///         }
+    ///         Event::Eof => break,
+    ///         _ => (),
+    ///     }
+    /// }
+    /// assert_eq!(count, 3);
+    /// assert_eq!(txt, vec!["Test".to_string(), "Test 2".to_string()]);
+    /// ```
+    ///
+    /// [`event_namespace()`]: Self::event_namespace
+    /// [`read_event_ns()`]: Self::read_event_ns
+    #[inline]
+    pub fn read_event(&mut self) -> Result<Event<'i>> {
+        self.read_event_impl(())
+    }
+
+    /// Reads the next event, borrow its content from the input buffer, and resolves
+    /// its namespace (if applicable).
+    ///
+    /// Namespace is resolved only for [`Start`], [`Empty`] and [`End`] events.
+    /// For all other events the concept of namespace is not defined, so
+    /// a [`ResolveResult::Unbound`] is returned.
+    ///
+    /// If you are not interested in namespaces, you can use [`read_event()`]
+    /// which will not automatically resolve namespaces for you.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use pretty_assertions::assert_eq;
+    /// use quick_xml::NsReader;
+    /// use quick_xml::events::Event;
+    /// use quick_xml::name::{Namespace, QName, ResolveResult::*};
+    ///
+    /// let mut reader = NsReader::from_str(r#"
+    ///     <x:tag1 xmlns:x="www.xxxx" xmlns:y="www.yyyy" att1 = "test">
+    ///        <y:tag2><!--Test comment-->Test</y:tag2>
+    ///        <y:tag2>Test 2</y:tag2>
+    ///     </x:tag1>
+    /// "#);
+    /// reader.trim_text(true);
+    ///
+    /// let mut count = 0;
+    /// let mut txt = Vec::new();
+    /// loop {
+    ///     match reader.read_event_ns().unwrap() {
+    ///         (Bound(Namespace(b"www.xxxx")), Event::Start(e)) => {
+    ///             count += 1;
+    ///             assert_eq!(e.local_name(), QName(b"tag1").into());
+    ///         }
+    ///         (Bound(Namespace(b"www.yyyy")), Event::Start(e)) => {
+    ///             count += 1;
+    ///             assert_eq!(e.local_name(), QName(b"tag2").into());
+    ///         }
+    ///         (_, Event::Start(_)) => unreachable!(),
+    ///
+    ///         (_, Event::Text(e)) => {
+    ///             txt.push(e.decode_and_unescape(&reader).unwrap().into_owned())
+    ///         }
+    ///         (_, Event::Eof) => break,
+    ///         _ => (),
+    ///     }
+    /// }
+    /// assert_eq!(count, 3);
+    /// assert_eq!(txt, vec!["Test".to_string(), "Test 2".to_string()]);
+    /// ```
+    ///
+    /// [`Start`]: Event::Start
+    /// [`Empty`]: Event::Empty
+    /// [`End`]: Event::End
+    /// [`read_event()`]: Self::read_event
+    #[inline]
+    pub fn read_event_ns(&mut self) -> Result<(ResolveResult, Event<'i>)> {
```

Comment:
I'm not sure this naming is an improvment, it feels too brief.  Perhaps `read_event_with_ns()` would be better.

Reply:
I agree, not a better name. May be `read_resolved_event`?

Reply:
Changed to `read_resolved_event` / `read_resolved_event_into`